### PR TITLE
Feat/list options

### PR DESCRIPTION
**Pull request by pedrobslisboa**
# Issue
#35

## What it does
Add questions and `figlet` to project

## Before 
 The only way to use it was by declaring `stack` and `filename`
```sh
npx @significa/start next hello-world
```

## After
We can use 
```sh
npx @significa/start next hello-world
```

And we have questions too

Captura de tela de 2020-07-28 16-29-27
Captura de tela de 2020-07-21 09-07-03

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -33,16 +35,57 @@ class SignificaStart extends Command {
...
       name: 'type',
-      required: true,
+      required: false,
       options: (stacks as unknown) as string[],
     },
-    { name: 'name', required: true },
+    { name: 'name', required: false },
+  ]
+
+  questions = [
```
Comment (danilowoz) [inline src/index.ts:44]: Shouldn't be a static method like the one above?

File: src/index.ts [hunk 2]
```diff
@@ -33,16 +35,57 @@ class SignificaStart extends Command {
...
+    )
+
+    const { args }: { args: { type: Stacks; name: string } } = this.parse(
+      SignificaStart
+    )
+
+    let type = args.type
+    let name = args.name
+
+    if (!name && !type) {
```
Comment (danilowoz) [inline src/index.ts:86]: I guess the right condition would be if the `name` OR `type` weren't provided, make sense?

File: src/index.ts [hunk 3]
```diff
@@ -33,16 +35,57 @@ class SignificaStart extends Command {
...
-    { name: 'name', required: true },
+    { name: 'name', required: false },
+  ]
+
+  questions = [
+    {
+      message: 'Which stack would you like to use?',
+      type: 'list',
+      name: 'type',
+      choices: [
```
Comment (pbrandone) [inline src/index.ts:49]: Please use this order:
- NextJS
- Gatsby
- CRA

File: src/index.ts [hunk 4]
```diff
@@ -33,16 +35,57 @@ class SignificaStart extends Command {
...
+          value: 'next',
+        },
+        {
+          name: 'Gatsby',
+          value: 'gatby',
+        },
+      ],
+    },
+    {
+      message: "What is the project's name?",
```
Comment (pbrandone) [inline src/index.ts:65]: Copy:

`What's the name of the project?`

File: src/index.ts [hunk 5]
```diff
@@ -33,16 +35,57 @@ class SignificaStart extends Command {
...
+      `\n${chalk.yellow(
+        figlet.textSync('significa-start', { horizontalLayout: 'default' })
+      )}`
+    )
+
+    const { args }: { args: { type: Stacks; name: string } } = this.parse(
+      SignificaStart
+    )
+
+    let type = args.type
```
Comment (pbrandone) [inline src/index.ts:83]: maybe 
```jsx
const name = args.name || await prompt({ message: "What's the name of the project", type: 'input', name: 'name', default: 'hello-world' }).name
```

and the same for type.
would this work?

Comment (pedrobslisboa) [reply src/index.ts:83]: It works :+1:

File: src/index.ts (hunk 2 above)
Comment (pedrobslisboa) [reply src/index.ts:86]: Perfect, totally agree.

File: src/index.ts (hunk 1 above)
Comment (pedrobslisboa) [reply src/index.ts:44]: I was thinking about it, but reading the other comments I've decided to use like this, with the question on method call:

```js
    const type =
      args.type ||
      (
        await inquirer.prompt({
          message: 'Which stack would you like to use?',
          type: 'list',
          name: 'type',
          choices: [
            {
              name: 'Create React App',
              value: 'cra',
            },
            {
              name: 'Nextjs',
              value: 'next',
            },
            {
              name: 'Gatsby',
              value: 'gatby',
            },
          ],
        })
      ).type

    const name =
      args.name ||
      (
        await inquirer.prompt({
          message: "What is the project's name?",
          type: 'input',
          name: 'name',
          default: 'hello-world',
        })
      ).name
```
I guess it's better to understand which variable is for which question.
What about it?

Comment (danilowoz) [reply src/index.ts:44]: Looks good to me

**Pull request merged**